File: app/src/main/java/alex/com/githubchecker/components/app/dagger/AppModule.java
package alex.com.githubchecker.components.app.dagger;

import android.content.Context;

import javax.inject.Inject;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Alex on 11/11/2017.
 */

@Module
public class AppModule {

    Context appContext;

    @Inject
    public AppModule(Context context) {
        appContext = context;
    }

    @Provides
    Context provideContext() {
        return appContext;
    }

}

File: app/src/main/java/alex/com/githubchecker/components/app/dagger/AppModule.kt
package alex.com.githubchecker.components.app.dagger

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Inject

/**
 * Created by Alex on 11/11/2017.
 */

@Module
class AppModule @Inject
constructor(val application: Application) {

    @Provides
    internal fun provideApplication(): Application {
        return application
    }

    @Provides
    internal fun provideContext(): Context {
        return application.applicationContext
    }
}

